docs/providers/datadog.md

```diff
@@ -0,0 +1,155 @@
+# Datadog
+
+## Backend
+
+Using the `Datadog` backend class, you can query any metrics available in
+Datadog to create an SLO.
+
+The following methods are available to compute SLOs with the `Datadog`
+backend:
+
+* `good_bad_ratio` for computing good / bad metrics ratios.
+* `query_sli` for computing SLIs directly with Datadog.
+* `query_slo` for getting SLO value from Datadog SLO endpoint.
+
+### Good / bad ratio
+
+The `good_bad_ratio` method is used to compute the ratio between two metrics:
+
+- **Good events**, i.e events we consider as 'good' from the user perspective.
+- **Bad or valid events**, i.e events we consider either as 'bad' from the user
+perspective, or all events we consider as 'valid' for the computation of the
+SLO.
+
+This method is often used for availability SLOs, but can be used for other
+purposes as well (see examples).
+
+**Config example:**
+
+```yaml
+backend:
+  class:   Datadog
+  method:  good_bad_ratio
+  api_key: ${DATADOG_API_KEY}
+  app_key: ${DATADOG_APP_KEY}
+  measurement:
+    filter_good: sum:system.disk.used{*}
```

Comment:
I don't think disk usage is a good example for an SLO. We encourage people to create SLOs based upon symtoms: things that are actually impacting users. This feels more like it's encouraging a cause-based approach.

Reply:
Done